### PR TITLE
Poprawki do zadania drugiego

### PR DESCRIPTION
**Pull request by bartekryba**
Rozwinięte komentarze, dodane testy, zapuszczony black na kodzie.

### CODE REVIEW

**Review by war-in — APPROVED** [review]
Does every method have its own test (if possible)?

File: exercise2/transaction_registry.py [hunk 1]
```diff
@@ -5,14 +5,16 @@
 
 class Transaction:
     """
-    K:
+    Transakcja zawiera:
     - odbiorcę transakcji (klucz publiczny)
     - hash poprzedniej transakcji
+    - wlasny hash, wyliczony na podstawie dwóch pól powyżej
```
Comment (war-in) [inline exercise2/transaction_registry.py:11]:
```suggestion
    - własny hash, wyliczony na podstawie dwóch pól powyżej
```

File: exercise2/transaction_registry.py [hunk 2]
```diff
@@ -5,14 +5,16 @@
...
+    - opcjonalny podpis
     """
 
     recipient: PublicKey
-    previous_tx_hash: bytes
-    tx_hash: bytes
+    previous_hash: bytes
```
Comment (war-in) [inline exercise2/transaction_registry.py:16]: Actually `previous_tx_hash` was not that bad. I would leave it as it was

**Review by war-in — APPROVED** [review]

File: exercise2/transaction_registry.py [hunk 3]
```diff
@@ -1,53 +1,49 @@
...
+        self.hash = hash(self.recipient.to_bytes() + self.previous_tx_hash)
 
-    @staticmethod
-    def from_transaction(transaction: Transaction, signature: bytes):
-        return SignedTransaction(transaction.recipient, transaction.previous_tx_hash, signature)
+    def sign(self, private_key: PrivateKey):
+        """
+        Podpisuje transakcję przy pomocy podanego klucza prywatnego.
+        """
+        self.signature = sign(private_key, self.hash)
```
Comment (MaksymilianDemitraszek) [inline exercise2/transaction_registry.py:35]: może warto dodać metodę `is_signed -> bool`

**Review by MaksymilianDemitraszek — COMMENTED** [review]
LGTM

**Pull request merged**